src/models/quote.js

```diff
@@ -1,38 +1,29 @@
 'use strict'
 
-const request = require('co-request')
 const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
 const UnacceptableAmountError = require('../errors/unacceptable-amount-error')
 const AssetsNotTradedError = require('../errors/assets-not-traded-error')
-const ExternalError = require('../errors/external-error')
 const DEFAULT_DESTINATION_EXPIRY = 5 // seconds
 
 function * makeQuoteQuery (params) {
-  const sourceLedger = params.source_ledger ||
-    (yield getAccountLedger(params.source_account))
-  const destinationLedger = params.destination_ledger ||
-    (yield getAccountLedger(params.destination_account))
+  const sourceLedger = addressToLedger(params.source_address)
+  const destinationLedger = addressToLedger(params.destination_address)
   const explain = params.explain === 'true'
   return {
     sourceLedger,
     destinationLedger,
     sourceAmount: params.source_amount,
     destinationAmount: params.destination_amount,
+    destinationPrecisionAndScale: params.destination_precision && {
+      precision: params.destination_precision,
+      scale: params.destination_scale
+    },
     explain
   }
 }
 
-function * getAccountLedger (account) {
-  const res = yield request({
-    method: 'get',
-    uri: account,
-    json: true
-  })
-  const ledger = res.body && res.body.ledger
-  if (res.statusCode !== 200 || !ledger) {
-    throw new ExternalError('Unable to identify ledger from account: ' + account)
-  }
-  return ledger
+function addressToLedger (address) {
+  return address.split('.').slice(0, -1).join('.')
```

Comment:
Ok, so if I'm understanding correctly, the actual money stops flowing once it hits the `us.fed.wf.stefan` account, so for example a transfer from `us.fed.wf.stefan.phone` to `us.fed.wf.stefan.laptop` would not have any effect on the balance at `us.fed.wf.stefan` and thus isn't transferring any real money in the way that a transfer from `us.fed.wf.stefan` to `us.fed.boa.stefan` does (which requires counterparties). It seems like a significant semantic difference that the address notation abstracts. Does this abstraction provide benefits or does it potentially obfuscate the actual semantics? Comparing with gmail addresses, `account+alias@domain` (comparable to `stefan+phone@wf.fed.us`) there is a syntactic indicator for each semantic category. Since this is a fairly big decision, I think it might be worthwhile to have a design rationale to justify why our solution is preferable to the more conventional approach. Also, there are still outstanding questions on the original issue which may affect this pull request: https://github.com/interledger/rfcs/issues/31